File: Arrays/Medium/Max Consecutive Ones.java
// ques link : https://leetcode.com/problems/max-consecutive-ones/
// TC O(n)
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        
        
        int curMax = 0;
        int max = 0;
        
        
        for(int a : nums){
            
            if(a == 1) curMax++;
            
            else {
                
                max = Math.max(curMax , max);
                curMax = 0;
            }
        }
        
        max = Math.max(max , curMax);
        
        
        return max;
    }
}

File: Arrays/Medium/Max Consecutive Ones.cpp
// ques link : https://leetcode.com/problems/max-consecutive-ones/
// TC O(n)
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int currentMax = 0 , finalMax = 0;
        for(auto x : nums)
        {
            if( x == 1)
                currentMax++;
            else
            {
                finalMax = max(currentMax , finalMax);
                currentMax = 0;
            }
        }
        
        finalMax = max(currentMax , finalMax);
        return finalMax;
    }
};